### PR TITLE
1.2 - GEOMESA-1653 Not always returning default geometry if alternate is re…

### PR DESCRIPTION
**Pull request by elahrvivaz**
…quested

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala [hunk 1]
```diff
@@ -215,7 +213,7 @@ object QueryPlanner extends LazyLogging {
   }
 
   def setPerThreadQueryHints(hints: Map[AnyRef, AnyRef]): Unit = threadedHints.put(hints)
-  def clearPerThreadQueryHints() = threadedHints.clear()
+  def clearPerThreadQueryHints(): Unit = threadedHints.clear()
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala]: Is this type annotation necessary?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala [hunk 2]
```diff
@@ -330,7 +328,7 @@ object QueryPlanner extends LazyLogging {
     * @param sft simple feature type
     * @return
     */
-  def setQueryTransforms(query: Query, sft: SimpleFeatureType) = {
+  def setQueryTransforms(query: Query, sft: SimpleFeatureType): Unit = {
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala]: Ditto

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala (hunk 1 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala]: haha, intellij has started flagging public methods without explicit return types, so I've started filling them in :P

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala [hunk 3]
```diff
@@ -240,5 +240,47 @@ class AccumuloDataStoreTransformsTest extends Specification with TestWithMultipl
...
+      "if default geometry is specified" >> {
+        val query = new Query(sftName, Filter.INCLUDE, Array("name", "geom"))
+        val features = SelfClosingIterator(ds.getFeatureSource(sftName).getFeatures(query).features).toSeq
+        features must haveSize(3)
+        forall(features)(_.getAttributeCount mustEqual 2)
+        features.map(_.getAttribute("geom")) must containTheSameElementsAs(reference.map(_.getAttribute("geom")))
+        features.map(_.getAttribute("name")) must containTheSameElementsAs(reference.map(_.getAttribute("name")))
+      }
+
+      "if alternate geometry is specified" >> {
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: Should we have a test for when a geometry attribute is renamed via a transform?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: i can add one

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: i added a test, but it fails - there isn't really an easy way to determine what is going to be returned from a transform expression. I tried using the `Definition.binding`, but that doesn't get populated. I think for now I'll leave it as is...

Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: Ah, makes sense; is it worth entering a ticket and/or adding the failing test with a pendingUntilFixed?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: dunno, think we'll ever fix it?

Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: What happens if you use the transform?  Does it work or throw exceptions?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTransformsTest.scala]: it works, but returns the default geometry. i'll add a failing unit test in the 1.3 branch

**Pull request merged**